kubric/viewer/blender.py

```diff
@@ -335,9 +363,30 @@ def __init__(self, geometry: Geometry, material: Material):
       self._blender_object.data.from_pydata(vertices, [], faces)
 
     # --- Adds the material to the object
+    # TODO: why not just use active_material as in set_material method?
```

Comment:
Blender allows multiple materials per object, for example to assign different materials to different parts of the mesh. 
If I understand correctly, then `materials.append` adds a new slot and fills it with the material. If the object did not have a material before, then this is also assigning the material to the entire mesh. However subsequent `append`s will only add material slots and not assign them (meaning that there is no visible change).
Setting `active_material` changes the active slot to a particular material. If the active slot is assigned to the mesh, then this will change the visible material of the object. 

So in short: Both versions will work if only called once on an object with no material. Setting `active_material` will also work if the object has a single material, or if the right slot is active. A fully general solution would either have to clear the slots first, or explicitly assign the material to the entire mesh.